flaky: templateversions unexpected status code 404: No previous template version

Spotted in: https://github.com/coder/coder/actions/runs/3631609128/jobs/6128439902

```
    templateversions_test.go:996: 
        	Error Trace:	C:\a\coder\coder\coderd\templateversions_test.go:996
        	Error:      	Received unexpected error:
        	            	GET http://localhost:53835/api/v2/organizations/a68d2356-90a4-461b-8a60-39deeceb5370/templateversions/dreamy_blackburn0/previous: unexpected status code 404: No previous template version found for "dreamy_blackburn0".
        	Test:       	TestPreviousTemplateVersion/Previous_version_found
    stuntest.go:64: STUN server shutdown
    --- FAIL: TestPreviousTemplateVersion/Previous_version_found (0.35s)
```

--- a/coderd/templateversions_test.go
+++ b/coderd/templateversions_test.go
@@ -968,9 +968,10 @@ func TestPreviousTemplateVersion(t *testing.T) {
 	t.Parallel()
 	t.Run("Previous version not found", func(t *testing.T) {
 		t.Parallel()
-		client := coderdtest.New(t, nil)
+		client := coderdtest.New(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
 		user := coderdtest.CreateFirstUser(t, client)
 		version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, nil)
+		coderdtest.AwaitTemplateVersionJob(t, client, version.ID)
 
 		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
 		defer cancel()
@@ -983,11 +984,13 @@ func TestPreviousTemplateVersion(t *testing.T) {
 
 	t.Run("Previous version found", func(t *testing.T) {
 		t.Parallel()
-		client := coderdtest.New(t, nil)
+		client := coderdtest.New(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
 		user := coderdtest.CreateFirstUser(t, client)
 		previousVersion := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, nil)
+		coderdtest.AwaitTemplateVersionJob(t, client, previousVersion.ID)
 		template := coderdtest.CreateTemplate(t, client, user.OrganizationID, previousVersion.ID)
 		latestVersion := coderdtest.UpdateTemplateVersion(t, client, user.OrganizationID, nil, template.ID)
+		coderdtest.AwaitTemplateVersionJob(t, client, latestVersion.ID)
 
 		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
 		defer cancel()

